fdb-relational-server/src/main/java/com/apple/foundationdb/relational/server/FRL.java

```diff
@@ -371,6 +371,16 @@ public void transactionalRollback(TransactionalToken token) throws SQLException
         token.getConnection().rollback();
     }
 
+    public void enableAutoCommit(TransactionalToken token) throws SQLException {
+        // we don't actually call setAutoCommit(false) until an operation happens, so if there is no token
```

Comment:
I think that - aligning this towards the future, where the client connection state will be in sync with the server-side connection state (TransactionalRequestHandler) - the logic of looking at the token should reside in the TransactionalRequestHandler. 
FRL would be a simpler, stateless pass through to the EmbeddedConnection.